src/server/server_family.cc

```diff
@@ -817,6 +823,91 @@ void SetSlowLogThreshold(util::ProactorPool& pool, int32_t val) {
   });
 }
 
+absl::flat_hash_map<std::string, std::string> GetCertInfo() {
+  absl::flat_hash_map<std::string, std::string> cert_info;
+  std::ifstream file(absl::GetFlag(FLAGS_tls_cert_file));
+  if (!file.is_open()) {
+    // cert_info.push_back("Error opening certificate file");
+    return cert_info;
+  }
+
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+  std::string cert_str = buffer.str();
+  const char* cert_cstr = cert_str.c_str();
+
+  BIO* bio = BIO_new_mem_buf(cert_cstr, -1);
+  if (!bio) {
+    // cert_info.push_back("Error creating BIO");
+    return cert_info;
+  }
+
+  X509* cert = PEM_read_bio_X509(bio, nullptr, nullptr, nullptr);
```

Comment:
Thank you for suggesting the solution, a couple of comments.

1. We usually define a struct with named variables in C++, instead of returning dictionaries.
2. It's better to fetch the certificate from the `SSL_CTX` object, using SSL_CTX_get0_certificate